docs/docs/features/splits.mdx

```diff
@@ -0,0 +1,180 @@
+    A[Computer/Host Device] <-- USB or BLE --> DONGLE[Dongle/Central];
+    DONGLE -. BLE .-> LEFT[Left Half/Peripheral];
+    DONGLE -. BLE .-> RIGHT[Right Half/Peripheral];
+```
+
+Dotted lines represent BLE connections.
+Solid lines represent USB connections. */}
+
+:::info
+For more information on how to set up a two split keyboard, refer to [New Keyboard Shield](../development/new-shield.mdx) under Guides.
+After setting up the split keyboard, you can follow the steps below to add a dongle to the setup.
+:::
+
+## Benefits/Disadvantages of using a Dongle
+
+Benefits:
+
+- Adding a dongle, enables both keyboard halves to act as peripherals, meaning equal battery life for both halves.
+- Easier to connect to the host device.
+
+Disadvantages:
+
+- More complex setup.
+- Extra microcontroller needed
+- Keyboard is unusable without the dongle.
+- Added latency due to the extra hop between the host device and the keyboard halves.
+
+
+
+## Defining a Dongle for a Keyboard 
+
+
+:::note
+In the examples below, we will be referencing the New Split Keyboard Shield guide files and will be building on top of that.
+:::
+
+### Keyboard Shield
+
+First we will introduce a third split to our keyboard configuration. This will be used as the dongle.
+
+```kconfig title="Kconfig.shield"
+config SHIELD_MY_BOARD_DONGLE
+    def_bool $(shields_list_contains,my_board_dongle)
+
+config SHIELD_MY_BOARD_LEFT
+    def_bool $(shields_list_contains,my_board_left)
+
+config SHIELD_MY_BOARD_RIGHT
+    def_bool $(shields_list_contains,my_board_right)
+```
+
+### Keyboard Defconfig
+
+Next we will define the roles of the keyboard halves. The left and right halves will be set as peripherals, and the dongle will be set as the central.
+We will also only give the central device the keyboard name.
+
+For the dongle configuration we also want to specify that we are expecting 2 peripherals, one for the left half and one for the right half.
+We will also set the `ZMK_SLEEP` to `n` to prevent the dongle from going to sleep. 
+If the dongle is sleeping, keystrokes will not be sent to the host device, until the dongle wakes up.
+
+Finally, we will set the `BT_CLTR_TX_PWR_PLUS_8` to `y` to increase the transmit power of the dongle.
+Normally this would drain the battery faster, but since the dongle is connected to the host device, it will not be a problem.
+
+There is also an option to still use the left half as the central, and the right half as the peripheral. In that case the dongle will not be used.
+
+
+```kconfig title="Kconfig.defconfig"
+if SHIELD_MY_BOARD_DONGLE
+
+config ZMK_KEYBOARD_NAME
+    default "My Board"
+
+config ZMK_SPLIT_ROLE_CENTRAL
+    default y
+
+config ZMK_SPLIT_BLE_CENTRAL_PERIPHERALS
+    default 2
+
+config ZMK_SLEEP
+    default n
+
+config BT_CLTR_TX_PWR_PLUS_8
+    default y
+
+endif
+
+if SHIELD_MY_BOARD_LEFT
+# Setting this to y will make the left half the central and the right half the peripheral
+config ZMK_SPLIT_ROLE_CENTRAL
+    default n
```

Comment:
Right, so my idea was to mark this as defaulting to `y`. The idea is that you could add all of this dongle setup as part of a ZMK module defining the shield. Then if another user wants to use the keyboard, they import the module, add their `config/my_board.keymap`, and add

```yaml
- board: microcontroller
  shield: my_board_left
- board: microcontroller
  shield: my_board_left
```
to their `build.yaml`. If said user then wants to use a dongle with `my_board`, all they have to add is

```kconfig
CONFIG_ZMK_SPLIT_ROLE_CENTRAL=n
```
to a file called `my_board_left.conf`, and add
```yaml
- board: microcontroller
  shield: my_board_dongle
```
to their `build.yaml`. I think we should be encouraging a method of adding dongles that interacts nicely with the modules system and doesn't require every user to redefine the keyboard just to add a dongle. A comment reminding users of the final line in [this section](https://zmk.dev/docs/config#user-config-folder) would be good for this.

Reply:
I think I need you to write this out for me, I am a bit confused on the implementation

Reply:
Improved